src/synth/synth-stmts.adb

```diff
@@ -1925,6 +1925,10 @@ package body Synth.Stmts is
 
       --  Handle the condition as an if.
       Cond := Get_Condition_Clause (Stmt);
+      if Cond = 0 then
```

Comment:
Please, use Null_Node instead of 0.
(I have to investigate why you could use 0 here, that's not expected).